lib/crawler/output_sink/elasticsearch.rb

```diff
@@ -63,19 +68,21 @@ def close
         system_logger.info(msg)
       end
 
-      def flush # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
-        payload = operation_queue.pop_all
-        if payload.empty?
-          system_logger.debug('Queue was empty when attempting to flush.')
+      def process # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        lock_queue
```

Comment:
I'd probably move the lock/unlock into its own context function, scary stuff should live alone

Reply:
I don't think this applies anymore as I've switched to `mutix.synchronize` with a timeout.